### PR TITLE
Streaming endpoint

### PR DESCRIPTION
**Pull request by NahuelBerg**
This branch has inside all the functionality added in the following tasks:
- SAUL-61
- SAUL-62
- SAUL-63
- SAUL-64

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Review by Renkon — APPROVED** [review]
LGTM

**Review by damatojuan — CHANGES_REQUESTED** [review]
image

It seems scale-unit parameter description has a wrong 'ScaleUnit' label at the beginning.

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_streaming_endpoint_scenarios.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        self.cmd('az ams account create -n {amsname} -g {rg} --storage-account {storageAccount} -l {location}')
+
+        self.cmd('az ams streaming endpoint create -g {rg} -a {amsname} -n {streamingEndpointName} -l {location} --availability-set-name {availabilitySetName} --cdn-provider {cdnProvider} --cdn-profile {cdnProfile} --description "{description}" --max-cache-age {maxCacheAge} --scale-units {scaleUnits} --tags "{tags}" --client-access-policy "{clientAccessPolicy}" --cross-domain-policy "{crossDomainPolicy}"', checks=[
+            self.check('scaleUnits', 4)
+        ])
+
+        self.cmd('az ams streaming endpoint scale -g {rg} -a {amsname} -n {streamingEndpointName} --scale-unit 10')
+
+        self.cmd('az ams streaming endpoint show -g {rg} -a {amsname} -n {streamingEndpointName}', checks=[
+            self.check('scaleUnits', 10)
```
Comment (damatojuan) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_streaming_endpoint_scenarios.py]: You shouldn't hardcode this value. Instead, use the scaleUnits property in the kwargs collection and check with the placeholder.

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_streaming_endpoint_scenarios.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+            'scaleUnits': 4,
+            'tags': 'foo=bar',
+            'clientAccessPolicy': self._get_test_data_file('clientAccessPolicy.xml'),
+            'crossDomainPolicy': self._get_test_data_file('crossDomainPolicy.xml')
+        })
+
+        self.cmd('az ams account create -n {amsname} -g {rg} --storage-account {storageAccount} -l {location}')
+
+        self.cmd('az ams streaming endpoint create -g {rg} -a {amsname} -n {streamingEndpointName} -l {location} --availability-set-name {availabilitySetName} --cdn-provider {cdnProvider} --cdn-profile {cdnProfile} --description "{description}" --max-cache-age {maxCacheAge} --scale-units {scaleUnits} --tags "{tags}" --client-access-policy "{clientAccessPolicy}" --cross-domain-policy "{crossDomainPolicy}"', checks=[
+            self.check('scaleUnits', 4)
```
Comment (damatojuan) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_streaming_endpoint_scenarios.py]: You shouldn't hardcode this value. Instead, use the scaleUnits property in the kwargs collection and check with the placeholder.

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_streaming_endpoint_scenarios.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        ])
+
+        self.cmd('az ams streaming endpoint scale -g {rg} -a {amsname} -n {streamingEndpointName} --scale-unit 10')
```
Comment (damatojuan) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_streaming_endpoint_scenarios.py]: You shouldn't hardcode this value. Instead, use the scaleUnits property in the kwargs collection.

**Review by damatojuan — CHANGES_REQUESTED** [review]

**Pull request merged**